pandas/tests/internals/test_external_block.py

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=W0102
+
+import numpy as np
+
+import pandas as pd
+from pandas.core.internals import Block, BlockManager
+
+
+class CustomBlock(Block):
+
+    def formatting_values(self):
+        return np.array(["Val: {}".format(i) for i in self.values])
+
+
+def test_custom_repr():
+    values = np.arange(3, dtype='int64')
+
+    # series
+    block = CustomBlock(values, placement=slice(0, 3))
```

Comment:
See here for a work around that I now use in the geopandas refactor branch: https://github.com/geopandas/geopandas/pull/467#issuecomment-319421369 
Basically it is constructing a dataframe, taking the blocks and axes out of it, append a Block to the blocks, and a value to the column axis, and create a new Block from those.

(but given the length of code in Block.insert/set, this is maybe actually a simple way)